### PR TITLE
README: document API differences when SQLite caching is enabled 

### PR DESCRIPTION
**Pull request by moio**
Part of https://github.com/rancher/rancher/issues/46409

This should help new developers being aware of any introduced differences.

### CODE REVIEW

**Comment by moio** [comment]
@richard-cox please let me know if any of this is a surprise to you (it really shouldn't, but it helps checking)

File: README.md [hunk 1]
```diff
@@ -117,13 +128,24 @@ Filters can be negated to exclude results:
...
+**If SQLite caching is disabled** (`server.Options.SQLCache=false`),
+arrays are searched for matching items. If any item in the array matches, the
 item is included in the list.
 
 ```
 /v1/{type}?filter=spec.containers.image=alpine
 ```
 
+**If SQLite caching is enabled** (`server.Options.SQLCache=true`),
+filtering is only supported for a subset of attributes:
```
Comment (richard-cox) [inline README.md:140]: this applies to sorting as well

File: README.md [hunk 2]
```diff
@@ -117,13 +128,24 @@ Filters can be negated to exclude results:
...
+- `metadata.name`, `metadata.namespace` and `metadata.timestamp` for any resource kind
+- a short list of hardcoded attributes for a selection of specific types listed
+in [typeSpecificIndexFields](https://github.com/rancher/steve/blob/main/pkg/stores/sqlproxy/proxy_store.go#L52-L58)
+- the special string `metadata.fields[N]`, with N starting at 0, for all columns
+displayed by `kubectl get $TYPE`. For example `secrets` have `"metadata.fields[0]"`,
```
Comment (richard-cox) [inline README.md:145]: i think these are crd additionalPrinterColumns (https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#additional-printer-columns), though not sure how those apply to built in resources. maybe too much detail for this doc?

**Review by richard-cox — COMMENTED** [review]
Not sure how much of a `how to` guide this should be, but might be worth guiding people to know ...
- cache off - native kube pagination (limit / continue)
- cache on - super awesome rancher pagination (pagesize, page)

File: README.md (hunk 1 above)
Comment (moio) [reply README.md:140]: nice catch, fixed in 3ce6f87

File: README.md (hunk 2 above)
Comment (moio) [reply README.md:145]: I am good with a bit more detail here, this is by devs for devs

Fixed in 2119538

**Comment by moio** [comment]
> Not sure how much of a how to guide this should be, but might be worth guiding people to know ...
> 
>    cache off - native kube pagination (limit / continue)
>    cache on - super awesome rancher pagination (pagesize, page)

Well, actually both sets of parameters work with both implementations! Their meaning is by and large comparable though.

**Review by richard-cox — APPROVED** [review]

**Pull request merged**